### PR TITLE
Add performance benchmarking scripts

### PR DESCRIPTION
**Pull request by rouault**
And run them by Travis-CI

Note: I've some doubt about the reproductibility and trustability in those tests on Travis infrastructure. Initially I had tried them on OSX workers too but this seemed far too unreliable : https://travis-ci.org/rouault/openjpeg/jobs/228744947

This relates to #911

### CODE REVIEW

File: tools/travis-ci/run.sh [hunk 1]
```diff
@@ -303,4 +305,43 @@ New/unknown test failure found!!!
...
+        echo "Comparing current version with Kakadu"
+        python compare_perfs.py /tmp/kakadu.csv /tmp/new.csv || true
+    fi
+    cd ..
+
+    REF_VERSION=master
+    if [ "${TRAVIS_PULL_REQUEST:-false}" == "false" ]; then
+        REF_VERSION=v2.1.2
+    fi
+    if [ ! -d ref_ojp ]; then
```
Comment (detonin) [inline tools/travis-ci/run.sh]: strangely enough, the correct prefix or suffix for OpenJPEG is opj and not ojp ... As everything in the code follows opj, I suggest we keep this rule.

File: tests/perf_test_filelist.csv [hunk 2]
```diff
@@ -0,0 +1,7 @@
+filename,iterations,threads,command,comment
+../data/input/nonregression/kodak_2layers_lrcp.j2c,3,1,DECOMPRESS,
+../data/input/nonregression/kodak_2layers_lrcp.j2c,5,2,DECOMPRESS,
+../data/input/nonregression/kodak_2layers_lrcp.j2c,10,4,DECOMPRESS,
+../data/input/conformance/p0_07.j2k,3,1,DECOMPRESS,128x128 tiles RLCP 3decomp 8layers no MCT reversible
+../data/input/conformance/p0_04.j2k,10,1,DECOMPRESS,precincts 128x128 irreversible
+../data/input/nonregression/X_4_2K_24_185_CBR_WB_000.tif,3,1,COMPRESS,
```
Comment (detonin) [inline tests/perf_test_filelist.csv]: do we need a newline here ?

**Comment by rouault** [comment]
> move performance-related files (scripts and filenames list to a "performance" folder in the "tests" directory)

Done

**Comment by rouault** [comment]
> instead of having the coding options hard-coded in the script, it would be useful to read an external file containing the options used in different environments

Agreed. I imagined that the COMPRESS string could be later extended to accept options like COMPRESS[reversible=yes,...] depending on what options we want to test.

File: tools/travis-ci/run.sh (hunk 1 above)
Comment (rouault) [inline tools/travis-ci/run.sh]: Done

File: tests/perf_test_filelist.csv (hunk 2 above)
Comment (rouault) [inline tests/perf_test_filelist.csv]: Done (wasn't strictly needed, but better indeed)

**Comment by rouault** [comment]
I've disabled making the job fail in case of performance regression, because even on Linux this is too unreliable. See https://travis-ci.org/uclouvain/openjpeg/jobs/229085228 that failed whereas previous builds (with same code) succeeded

**Comment by detonin** [comment]
> Agreed. I imagined that the COMPRESS string could be later extended to accept options like COMPRESS[reversible=yes,...] depending on what options we want to test.

ok. This way is possible indeed but would require a "translation" from this generic syntax to the actual kdu or opj options I guess.

> I've disabled making the job fail in case of performance regression, because even on Linux this is too unreliable. See https://travis-ci.org/uclouvain/openjpeg/jobs/229085228 that failed whereas previous builds (with same code) succeeded

ok. I think you're right : benchmark results on Travis will never be trustful. I wonder if we should not even disable the test but provide an easy way to run the script locally, as you suggested. To be discussed.
Do you already have results or graphs showing the current performances of OPJ vs KDU ?

**Comment by rouault** [comment]
> I wonder if we should not even disable the test but provide an easy way to run the script locally, as you suggested

The test can be run locally with
OPJ_SKIP_REBUILD=1 OPJ_CI_PERF_TESTS=1 OPJ_NONCOMMERCIAL=1 tools/travis-ci/run.sh

I think we can keep it on Travis in non-failing mode as I did in the latest version of the patch, so as a hint of possible regressions or improvements, to be confirmed locally.

> Do you already have results or graphs showing the current performances of OPJ vs KDU ?

Yes, extract from https://travis-ci.org/uclouvain/openjpeg/jobs/229431228
```
Comparing current version with Kakadu
../../data/input/nonregression/kodak_2layers_lrcp.j2c, 3 iterations, 1 threads, DECOMPRESS: ref_time 1830 ms, new_time 3196 ms, (regression) 74.6 %, ERROR_THRESHOLD
../../data/input/nonregression/kodak_2layers_lrcp.j2c, 5 iterations, 2 threads, DECOMPRESS: ref_time 1708 ms, new_time 4584 ms, (regression) 168.4 %, ERROR_THRESHOLD
../../data/input/nonregression/kodak_2layers_lrcp.j2c, 10 iterations, 4 threads, DECOMPRESS: ref_time 2138 ms, new_time 7831 ms, (regression) 266.3 %, ERROR_THRESHOLD
../../data/input/conformance/p0_07.j2k, 3 iterations, 1 threads, DECOMPRESS: ref_time 1948 ms, new_time 7418 ms, (regression) 280.8 %, ERROR_THRESHOLD
../../data/input/conformance/p0_04.j2k, 10 iterations, 1 threads, DECOMPRESS: ref_time 1096 ms, new_time 1270 ms, (regression) 15.9 %, ERROR_THRESHOLD
../../data/input/nonregression/X_4_2K_24_185_CBR_WB_000.tif, 3 iterations, 1 threads, COMPRESS: ref_time 4960 ms, new_time 9890 ms, (regression) 99.4 %, ERROR_THRESHOLD
TOTAL: ref_time 13683 ms, new_time 34192 ms, (regression) 149.9 %, ERROR_THRESHOLD
```

**Pull request merged**